src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -292,7 +292,11 @@ public static void ReThrowWithPath(in ReadStack state, JsonReaderException ex)
             string message = ex.Message;
 
             // Insert the "Path" portion before "LineNumber" and "BytePositionInLine".
+#if BUILDING_INBOX_LIBRARY
+            int iPos = message.AsSpan().LastIndexOf(" LineNumber: ");
```

Comment:
I wonder if this is overkill; this is an exception path. Perhaps just change the LOC below to use Ordinal?

Reply:
The problem I want to fix is that keeping a method like Something(StringComparison) internally ends up needing to handle any of the possible inputs, which means it roots the implementations for Ordinal{IgnoreCase}, InvariantCulture{IgnoreCase}, and CurrentCulture{IgnoreCase} and prevents them from being trimmed even if they're not used elsewhere (which most of them aren't).

Reply:
> As far as I can tell, currently the linker doesn't factor in const args to methods as part of rewriting the callee... @eerhardt, is that correct, and if so is there any plan for such an enhancement?

AFAIK, the linker doesn't do this level of analysis. It would involve multiple passes. We can ask the linker team (or log an issue in mono/linker) for it. But my assumption is that this would be rather costly (both development time, and time to run the linker).

Reply:
I opened https://github.com/mono/linker/issues/1868.